### PR TITLE
Support both operators and special callsigns on same instance

### PR DESCRIPTION
**Pull request by derFogel**
This patch adds a special auth_level for special callsings and only shows the operator popup if the user has the special_callsign auth_level.

### CODE REVIEW

File: application/config/config.sample.php
Comment (HB9HIL) [inline application/config/config.sample.php:77]: Please add this line in `install/config/config.php` aswell

**Review by HB9HIL — CHANGES_REQUESTED** [review]
I like the idea to make it more flexible. However I see two issues:

1. If we add a new Auth-Level in the sample config and install config. The new authentication level is see by ALL new instances. Even if they probably never use it. This can maybe result in wrong expectations?
Solution could be to do not add this at all to the configs and just adjust the wiki or make the wiki and config more clear with comments. Maybe @int2001 has an opinion about that. IMHO the `install/config/config.php` and `config.sample.php` should be identical in every case. 

2. A user with auth level "Special Callsign" has theoretically and practical a lower permission level then a regular "Operator" user. Therefore it would make more sense to give this user a permission level of `2` instead of `5`. The whole authentication checks in Wavelog are already checking for level `2` which wouldn't bring a big change (I think we have only three places in the API Controller which is checking for `3` but this can be changed). 


I agree with the fact, that this allows to have normal operators and special callsign on the same instance is a cool thing. But I would be careful how we deploy that. Because adding this to the config.sample.php is imho a big change which shows ALL new installs a third auth level in the user menu. Probably better solution would be to add just a tutorial / "How to activate" in the Wiki and do not add this in the config at all. **(Or add it but comment it out!)**

@int2001 @AndreasK79 @phl0 
I think we should discuss this in team how to handle that.

**Comment by HB9HIL** [comment]
@derFogel Please use your `dev` branch next time. Not your `master` branch
image

File: application/config/config.sample.php
Comment (derFogel) [reply application/config/config.sample.php:77]: `5` was more or less random. But will change that to `2`.
I like the idea with the comment, need to figure out how to edit the wiki once this is merged / released.

Comment (HB9HIL) [reply application/config/config.sample.php:77]: The Wiki is open to edit for everyone

https://github.com/wavelog/wavelog/wiki/Recommended-Setup-for-Special-Callsigns-and-Clubs

Comment (derFogel) [reply application/config/config.sample.php:77]: Ah that doesn't work. That would disable api access which would suck when you use wsjt-x.

Comment (HB9HIL) [reply application/config/config.sample.php:77]: @derFogel I don't see an issue to change API access to `authorize(2)` but the others have to agree with that

Comment (derFogel) [reply application/config/config.sample.php:77]: Actually the api (I tested with the radio api) works when I change it to 2

Comment (derFogel) [reply application/config/config.sample.php]: With a permission level of `2` you cannot log anything, so this will be a bigger change.

Comment (int2001) [reply application/config/config.sample.php]: my 2ct on that:
i like the idea to configure it as well.
but imho an option (via the user_options_model) may fit better here.

reasons:
- The grants (in a classical sense) are still the same for those stations.
- it's "only" the popup which is shown to those special-logins. logic is completly the same like a normal user has (as a "normal user" you can set the OP-Call at the first QSO - see station_tab. OP is persitant for the session
- with the user_option we don't touch the config.php - it is minimal-invasive compared to a new Userlevel

**Comment by HB9HIL** [comment]
Hello @derFogel 

I think we won't merge this one. We have other plans to address your request. I'm sorry. Please be patient. This will be addresses in the next bigger release (maybe 1.9 but I can't say for sure).

**Pull request closed without merging**
